project/src/main/webapp/shared.js

```diff
@@ -285,17 +315,31 @@ function buildSkillsAsLabels(querySelector, opportunities) {
 }
 
 /**
- * Adds currently-attributed profile image to logged-in user
- * If the user has no profile image, add the default one
+ * Adds currently-attributed image
+ * If no image, add a default image or background color with default height
  */
-async function populateExistingProfileImage() {
-  const handlerResponse = await fetch('/blob-handler');
-  const blobKey = await handlerResponse.text();
+async function populateExistingImage(type, elementId, eventId='') {
+  if (!eventId) {
+    eventId = getEventId();
+  }
+  const blobKey = await getBlobKey(type, eventId);
   let imageUrl = 'assets/default_profile.jpg';
-  if (blobKey.trim()) {
-    const serveResponse = await fetch(`/blob-serve?key=${blobKey}`);
+  const serveResponse = await fetch(`/blob-serve?key=${blobKey}`);
```

Comment:
maybe rename because every servlet response is a serveResponse so name is unclear

Reply:
Done